src/Fmt.ml

```diff
@@ -185,35 +188,96 @@ let debug_box_close fs =
         (fun fs -> Format.fprintf fs "@<0>]")
         fs )
 
-let open_box n fs = debug_box_open "b" n fs ; Format.pp_open_box fs n
+let open_box ?name n fs =
+  debug_box_open ?name "b" n fs ;
+  Format.pp_open_box fs n
 
-and open_vbox n fs = debug_box_open "v" n fs ; Format.pp_open_vbox fs n
+and open_vbox ?name n fs =
+  debug_box_open ?name "v" n fs ;
+  Format.pp_open_vbox fs n
 
-and open_hvbox n fs = debug_box_open "hv" n fs ; Format.pp_open_hvbox fs n
+and open_hvbox ?name n fs =
+  debug_box_open ?name "hv" n fs ;
+  Format.pp_open_hvbox fs n
 
-and open_hovbox n fs =
-  debug_box_open "hov" n fs ;
+and open_hovbox ?name n fs =
+  debug_box_open ?name "hov" n fs ;
   Format.pp_open_hovbox fs n
 
 and close_box fs = debug_box_close fs ; Format.pp_close_box fs ()
 
 (** Wrapping boxes ------------------------------------------------------*)
 
-let cbox n = wrap_k (open_box n) close_box
+module Safe = struct
+  type sep = t
+
+  let cut fs = Format.pp_print_cut fs ()
+
+  let space fs = Format.pp_print_space fs ()
+
+  let break n o fs = Format.pp_print_break fs n o
+
+  let linebreak o fs = Format.pp_print_break fs 1000 o
+
+  let double_linebreak fs = Format.fprintf fs "\n@;<1000 0>"
+
+  type boxed =
+    | Empty (* this value is not exposed to not allow empty boxes *)
+    | One of t
+    | Cons of boxed * sep * t
```

Comment:
This allows `Cons (Empty, sep t)`. `Empty` should be out of this type, eg. `type boxed' = One of t | Cons of ..` and `type boxed = boxed' option`
This causes an `impossible` in the code below.

Reply:
It should not happen since I didn't export the type `boxed` nor a way of construct `Empty` from the outside.

Reply:
`add (of_list [] noop ~f) s noop` wouldn't ?
Anyway, I meant that the representation allows it. Which make it less robust and you had to write `| Empty -> impossible "..."`.

Reply:
Yes I agree noop is problematic, but it has to be removed to not be allowed in these boxes, I will have a look at removing most of them, but maybe in another PR.